src/AltinnCore/Common/Services/Implementation/DataAppSI.cs

```diff
@@ -0,0 +1,147 @@
+                client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/xml"));
+
+                XmlSerializer serializer = new XmlSerializer(type);
+                using (MemoryStream stream = new MemoryStream())
+                {
+                    serializer.Serialize(stream, dataToSerialize);
+                    stream.Position = 0;
+                    StreamContent streamContent = new StreamContent(stream);
+                    streamContent.Headers.ContentType = MediaTypeHeaderValue.Parse("application/xml");
+                    Task<HttpResponseMessage> response = client.PostAsync(apiUrl, streamContent);
+                    if (!response.Result.IsSuccessStatusCode)
+                    {
+                        throw new Exception("Unable to save form model");
+                    }
+
+                    string instanceData = await response.Result.Content.ReadAsStringAsync();
+                    instance = JsonConvert.DeserializeObject<Instance>(instanceData);
+                }
+            }
+
+            return instance;
+        }
+
+        /// <summary>
+        /// update the form 
+        /// </summary>
+        /// <typeparam name="T">The input type</typeparam>
+        /// <param name="dataToSerialize">The data to serialize</param>
+        /// <param name="instanceId">The formId</param>
+        /// <param name="type">The type</param>
+        /// <param name="applicationOwnerId">The Organization code for the service owner</param>
+        /// <param name="applicationId">The service code for the current service</param>
+        /// <param name="instanceOwnerId">The partyId</param>
+        public void UpdateData<T>(T dataToSerialize, Guid instanceId, Type type, string applicationOwnerId, string applicationId, int instanceOwnerId, Guid dataId)
+        {
+            string apiUrl = $"{_platformStorageSettings.ApiUrl}/instances/{instanceId}/data/{dataId}?instanceOwnerId={instanceOwnerId}";
+
+            using (HttpClient client = new HttpClient())
+            {
+                client.BaseAddress = new Uri(apiUrl);
+                XmlSerializer serializer = new XmlSerializer(type);
+                using (MemoryStream stream = new MemoryStream())
+                {
+                    serializer.Serialize(stream, dataToSerialize);
+                    stream.Position = 0;
+                    StreamContent streamContent = new StreamContent(stream);
+                    streamContent.Headers.ContentType = MediaTypeHeaderValue.Parse("application/xml");
+                    Task<HttpResponseMessage> response = client.PutAsync(apiUrl, streamContent);
+                    if (!response.Result.IsSuccessStatusCode)
+                    {
+                        throw new Exception("Unable to save form model");
+                    }
+
+                    //return Guid.Parse(await response.Result.Content.ReadAsAsync<string>());
```

Comment:
should this be commented out?

Reply:
yes, as the method doenst return anyhting. I will remove the commented line